src/utilities/utilities.ts

```diff
@@ -321,3 +321,37 @@ export function expandFilePathTilda(filepath: string): string {
     }
     return filepath;
 }
+
+const regexEscapedCharacters = new Set(["(", ")", "[", "]", ".", "$", "^", "?", "|", "/", ":"]);
+/**
+ * Escapes regular expression special characters with a backslash.
+ * @param string A string to escape
+ * @returns The escaped string
+ */
+export function regexEscapedString(string: string): string {
+    let result = "";
+    for (const c of string) {
+        if (regexEscapedCharacters.has(c)) {
+            result += `\\${c}`;
+        } else {
+            result += c;
+        }
+    }
+    return result;
+}
+
+/**
+ * An implementation of `reduce()` that operates on a vscode.TestItemCollection,
+ * which only exposes `forEach` for iterating its items
+ */
+export function reduceTestItemChildren<U>(
```

Comment:
It's a little odd this is in utilities.ts. Is not one of the test explorer files better?